File: P1A11Termometro/C#/Termometro/Termometro.cs
using System;

class Termometro {

    // Atributo
    private float temperatura;

    // Constructor por defecto
    public Termometro() {
        Random rand = new Random();
        temperatura = (float)(rand.NextDouble() * 100); // El random va del 0 al 100
    }

    // Constructor parametrizado (recibe temperatura en °C)
    public Termometro(float temperatura) {
        this.temperatura = temperatura;
    }

    // Temperatura en Celcius
    public float temperaturaC() {
        return temperatura;
    }

    // Temperatura en F
    public float temperaturaF() {
        return (temperatura * 9 / 5) + 32;
    }

    // Temperatura en K
    public float temperaturaK() {
        return temperatura + 273.15f;
    }

    public void mostrartemperaturas() {
        Console.WriteLine("Temperatura en Celcius: " + temperaturaC());
        Console.WriteLine("Temperatura en Fahrenheit: " + temperaturaF());
        Console.WriteLine("Temperatura en Kelvin: " + temperaturaK());
    }

}

File: S2P1_setter_getter/P1A11Termometro/Python/termometro.py
import random

class Termometro:

    def __init__(self, temperatura=None):
        if temperatura is None:
            self.temperatura = random.uniform(0, 100)
        else:
            self.temperatura = temperatura

    def temperatura_c(self):
        return self.temperatura

    def temperatura_f(self):
        return (self.temperatura * 9/5) + 32

    def temperatura_k(self):
        return self.temperatura + 273.15

    def mostrar_temperaturas(self):
        print("Temperatura en Celsius:", self.temperatura_c())
        print("Temperatura en Fahrenheit:", self.temperatura_f())
        print("Temperatura en Kelvin:", self.temperatura_k())
